scripts/common.py

```diff
@@ -497,21 +497,32 @@ def is_port_in_use(port: int) -> bool:
         return bool(not s.connect_ex(('localhost', port)))
 
 
-def recursive_chown(path: str, uid: int, gid: int) -> None:
+def recursive_chown(
+    path: str, uid: Optional[int] = None, gid: Optional[int] = None
+) -> None:
     """Changes the owner and group id of all files in a path to the numeric
     uid and gid.
 
     Args:
         path: str. The path for which owner id and group id need to be setup.
         uid: int. Owner ID to be set.
         gid: int. Group ID to be set.
+
+    Raises:
+        Exception. Uid and gid are both None.
     """
-    os.chown(path, uid, gid)
-    for root, directories, filenames in os.walk(path):
-        for directory in directories:
-            os.chown(os.path.join(root, directory), uid, gid)
-        for filename in filenames:
-            os.chown(os.path.join(root, filename), uid, gid)
+    chown_command = 'chown -R '
+
+    if uid is not None and gid is not None:
+        chown_command += '%s:%s %s' % (uid, gid, path)
+    elif uid is not None:
+        chown_command += '%s %s' % (uid, path)
+    elif gid is not None:
+        chown_command += ':%s %s' % (gid, path)
+    else:
+        raise Exception(
+            'ERROR: Either uid or gid should be provided for chown.')
```

Comment:
Suggestion/Optional: 
```suggestion
    if uid is None and gid is None:
        raise ValueError('ERROR: Either uid or gid should be provided for chown.')

    owner_spec = ''
    if uid is not None:
        owner_spec += str(uid)
    if gid is not None:
        owner_spec += f':{gid}'
        
    chown_args = ['chown', '-R', owner_spec, path]
```

- Follows early-exist/gaurd clause principle. 
- Avoids to many connected if-else statements.